resource/readers/resource_reader_hwloc.cpp

```diff
@@ -274,6 +278,14 @@ int resource_reader_hwloc_t::walk_hwloc (resource_graph_t &g,
             type = "storage";
             basename = type;
             supported_resource = true;
+
+            const char* devID_str = hwloc_obj_get_info_by_name (obj, "LinuxDeviceID");
+            if (devID_str == NULL) {
+                // Non-fatal error, just log a warning
+                m_err_msg += "Error getting the LinuxDeviceID of block storage device; ";
+            } else {
+                properties["LinuxDeviceID"] = devID_str;
```

Comment:
It would be a bit safer with `insert`
```suggestion
                auto ret = properties.insert (
                               std::pair<std::string,
                                         std::string> ("LinuxDeviceID",
                                                       devID_str));
                if (!ret) {
                    errno = EEXIST;
                    m_err_msg += "Error inserting LinuxDeviceID into properties map; ";
                    rc = -1;
                    break;
                }